### DISCUSSION TITLE
Functions in Formulaic?

### DISCUSSION THREAD

**Discussion by johndburger** [CamDavidsonPilon/lifelines#1337] [Q&A]
I've searched this board and Formulaic's but can't figure this out.

This is my first time running some older code since Lifelines switched from Patsy. I'm using a formula with numpy's log10 function, like so:
`formula="... + np.log10(x) ..."`

Previously this worked fine, since the place where this is used does this:
`import numpy as np`

This no longer works, I get 
`NameError: name 'np' is not defined`

I've even tried importing `log10` directly and using this syntax:
`"... + I(log10(x)) ..."`

I still get:
`NameError: name 'log10' is not defined`

What am I missing, do I have to do something special to get Formulaic to understand such functions?

**Comment by CamDavidsonPilon** [comment]
Hi @johndburger, you're right, this is broken. I think I've identified what's going on, and am doing some tests now to see if I can fix it on my side

**Reply by CamDavidsonPilon** [reply]
Asking the Formulaic author [here](https://github.com/matthewwardrop/formulaic/issues/48)

**Reply by johndburger** [reply]
Excellent, thanks so much for your efforts!
